src/main/sensors/gyroanalyse.c

```diff
@@ -93,27 +91,20 @@ void initGyroData(void)
     }
 }
 
-static inline int fftFreqToBin(int freq)
-{
-    return ((FFT_WINDOW_SIZE / 2 - 1) * freq) / (fftMaxFreq);
-}
-
 void gyroDataAnalyseInit(uint32_t targetLooptimeUs)
 {
     // initialise even if FEATURE_DYNAMIC_FILTER not set, since it may be set later
-    samplingFrequency = 1000000 / targetLooptimeUs;
+    const uint16_t samplingFrequency = 1000000 / targetLooptimeUs;
     fftSamplingScale = samplingFrequency / FFT_SAMPLING_RATE;
-    fftMaxFreq = FFT_SAMPLING_RATE / 2;
-    fftBinCount = fftFreqToBin(fftMaxFreq) + 1;
-    fftResolution = FFT_SAMPLING_RATE / FFT_WINDOW_SIZE;
     arm_rfft_fast_init_f32(&fftInstance, FFT_WINDOW_SIZE);
 
     initGyroData();
     initHanning();
 
     // recalculation of filters takes 4 calls per axis => each filter gets updated every 3 * 4 = 12 calls
     // at 4khz gyro loop rate this means 4khz / 4 / 3 = 333Hz => update every 3ms
-    float looptime = targetLooptimeUs * 4 * 3;
+    // for gyro rate > 16kHz, we have update frequency of 1kHz => 1ms
+    const float looptime = MAX(1000000u / FFT_SAMPLING_RATE, targetLooptimeUs * 4 * 3);
```

Comment:
Let's extract this ratio to another preprocessor macro.
I would prefer the `4 * 3` factor to be extracted as well.